### PR TITLE
Replace rpm2cpio with rpm2archive

### PR DESCRIPTION
**Pull request by pmatilai**
Enhance rpm2archive to support cpio format too, replace rpm2cpio with it entirely.

This is important as rpm2cpio can't deal with 64bit file sizes which will  be the norm in v6.
For details, see the commit messages.

### CODE REVIEW

File: tools/rpm2archive.c [hunk 1]
```diff
@@ -141,10 +163,20 @@ static int process_package(rpmts ts, const char * filename)
...
+    if (rstreq(format, "pax")) {
+	format_code = ARCHIVE_FORMAT_TAR_PAX_RESTRICTED;
+    } else if (rstreq(format, "cpio")) {
+	format_code = ARCHIVE_FORMAT_CPIO_SVR4_NOCRC;
+    } else {
+	rc = ARCHIVE_FAILED;
+    }
+
+    if (archive_write_set_format(a, format_code) != ARCHIVE_OK) {
+	fprintf(stderr, "Error: Format %s is not supported\n", format);
```
Comment (dmnks) [inline tools/rpm2archive.c:176]: There seems to be no `format` variable, isn't this a forgotten "need to get the format name from `format_code`"?

Comment (dmnks) [reply tools/rpm2archive.c:176]: ... as I suppose using `format_code` wouldn't be optimal either as it's just an integer code.

**Review by dmnks — APPROVED** [review]
Looks good!

**Comment by pmatilai** [comment]
Okay, I guess we'll just merge then. Thanks for the eyeballs :smile:

**Pull request merged**